e2e/cypress/integration/multi_team_and_dm/max_gm_members_spec.js

```diff
@@ -0,0 +1,126 @@
+ * @param {number} expectedUsersLeftToAdd
+ */
+const expectCannotAddUsersMessage = (expectedUsersLeftToAdd) => {
+    const maxUsersGMNote = "You've reached the maximum number of people for this conversation. Consider creating a private channel instead.";
+
+    // * Check that the help section indicates we cannot add anymore users
+    cy.get('#multiSelectHelpMemberInfo').
+        should('be.visible').
+        and('contain.text', `You can add ${expectedUsersLeftToAdd} more ${expectedUsersLeftToAdd === 1 ? 'person' : 'people'}`);
+
+    if (expectedUsersLeftToAdd === 0) {
+        // * Check that a note in the help section suggests creating a private channel instead
+        cy.get('#multiSelectMessageNote').should('contain.text', maxUsersGMNote);
+    }
+};
+
+describe('Multi-user group messages', () => {
+    let testTeam;
+    before(() => {
+        // # Create a new team
+        cy.apiInitSetup().then(({team}) => {
+            testTeam = team;
+
+            // # Add 10 users to the team
+            Cypress._.times(10, createUserAndAddToTeam(testTeam));
+        });
+    });
+
+    it('MM-T463 Should not be able to create a group message with more than 8 users', () => {
+        // # Go to town-square channel
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Open the 'Direct messages' dialog to create a new direct message
+        cy.get('#addDirectChannel').click();
+
+        // # Add the maximum amount of users to a group message (7)
+        addUsersToGMViaModal(7);
+
+        // * Check that the user count in the group message modal equals 7
+        cy.get('.react-select__multi-value').should('have.length', 7);
+
+        expectCannotAddUsersMessage(0);
+
+        // # Try to add one more user
+        addUsersToGMViaModal(1);
+
+        // * Check that the list of users in the multi-select is still 7
+        cy.get('.react-select__multi-value').should('have.length', 7);
+
+        // # Click on "Go" in the group message's dialog to begin the conversation
+        cy.get('#saveItems').click();
+
+        // * Check that the number of users in the group message is 8
+        cy.get('#channelMemberCountText').
+            should('be.visible').
+            and('have.text', '8');
+
+        // # From the group message's window, click on the user list's dropdown
+        cy.get('#channelHeaderDropdownIcon').click();
+
+        // # From the dropdown menu, click on "Add members"
+        cy.get('#channelAddMembers').click();
+
+        // # Try to add one more user from the group message's list
+        addUsersToGMViaModal(1);
+
+        // # Check that the appropriate information & warnings are displayed in the group message's dialog
+        expectCannotAddUsersMessage(0);
+
+        // * Check that the count of selected number of users in the group message's dialogis is 7
+        cy.get('.react-select__multi-value').
```

Comment:
Not a fan of using classnames as selectors but ReactSelect does not seem to offer a way of setting HTML id's on these items.